runtime/JavaScript/src/antlr4/InputStream.d.ts

```diff
@@ -1,4 +1,12 @@
+/*
+ * Copyright (c) 2012-2017 The ANTLR Project. All rights reserved.
+ * Use of this file is governed by the BSD 3-clause license that
+ * can be found in the LICENSE.txt file in the project root.
+ */
+
 export declare class InputStream {
-    constructor(data: string);
-    constructor(data: string, decodeToUnicodeCodePoints: boolean);
+    public constructor(data: string);
```

Comment:
why does this constructor need to be public when the others don't ?

Reply:
If there's no access modifier then the default is public. But making it explicit which visibility a member has is considered good coding style. In JS it doesn't really matter, as there is no public or private way (the `#member' notation is an addendum to improve this situation), but for TS it makes sense as the transpiler enforces the correct access.

Reply:
"the transpiler enforces the correct access" -> you mean the type checker ? The transpiler drops this since it's not supported by JS.